### PR TITLE
[sizing cal][DB-12359] : impact of indexes on colocated tables import time

### PR DESCRIPTION
**Pull request by shaharuk-yb**
## Changes:
- added logic for improving import time for colocated tables based on impact of number of indexes
- removed old logic and associated functions
- added relevant unit tests
- added experiment data.

### table ```sharded_load_time_index_impact``` currently in main. 
Screenshot 2024-08-20 at 6 57 31 PM

### changes as part of this PR:  
- table renamed to ```load_time_index_impact``` 
- renamed column ```multiplication_factor``` to ```multiplication_factor_sharded ``` and inserted/updated values according to  latest available experiment data.
- added column ```multiplication_factor_colocated``` and inserted values accodring to latest available experiment data. Note that there are null values where indexes are more than 5 as any such tables would be sharded and hence not required.
Screenshot 2024-08-20 at 7 28 07 PM

### CODE REVIEW

File: yb-voyager/src/migassessment/sizing.go [hunk 1]
```diff
@@ -190,31 +194,40 @@ func SizingAssessment() error {
...
-	importTimeForColocatedObjects, parallelVoyagerJobsColocated, err :=
-		calculateTimeTakenAndParallelJobsForImportColocatedObjects(colocatedObjects,
-			finalSizingRecommendation.VCPUsPerInstance, finalSizingRecommendation.MemoryPerCore, experimentDB)
+	// get load times data from experimental database for sharded Tables
+	colocatedLoadTimes, err := getExpDataLoadTime(experimentDB, finalSizingRecommendation.VCPUsPerInstance,
+		finalSizingRecommendation.MemoryPerCore, COLOCATED_LOAD_TIME_TABLE)
 	if err != nil {
-		SizingReport.FailureReasoning = fmt.Sprintf("calculate time taken for colocated data import: %v", err)
-		return fmt.Errorf("calculate time taken for colocated data import: %w", err)
+		return fmt.Errorf("error while fetching sharded load time info: %w", err)
```
Comment (sonalsagarwal) [inline yb-voyager/src/migassessment/sizing.go:201]: should this error be for colocated table?

File: yb-voyager/src/migassessment/sizing.go [hunk 2]
```diff
@@ -190,31 +194,40 @@ func SizingAssessment() error {
...
+		finalSizingRecommendation.MemoryPerCore, SHARDED_LOAD_TIME_TABLE)
 	if err != nil {
 		return fmt.Errorf("error while fetching sharded load time info: %w", err)
 	}
 
-	// get experimental data for impact of indexes on sharded tables load
-	indexImpactData, err := getExpDataIndexImpactOnLoadTime(experimentDB, finalSizingRecommendation.VCPUsPerInstance, finalSizingRecommendation.MemoryPerCore)
+	// get experimental data for impact of indexes on colocated tables load
+	indexImpactOnLoadTimeCommon, err := getExpDataIndexImpactOnLoadTime(experimentDB,
+		finalSizingRecommendation.VCPUsPerInstance, finalSizingRecommendation.MemoryPerCore)
```
Comment (sonalsagarwal) [inline yb-voyager/src/migassessment/sizing.go:213]: is this same for colocated and sharded table?

**Review by sonalsagarwal — CHANGES_REQUESTED** [review]

File: yb-voyager/src/migassessment/sizing.go (hunk 1 above)
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/sizing.go]: done. made the changes.

File: yb-voyager/src/migassessment/sizing.go (hunk 2 above)
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/sizing.go:213]: done. made the changes.

**Review by sonalsagarwal — APPROVED** [review]
LGTM, will confirm db file data later

**Pull request merged**